### PR TITLE
Easy wasm builds

### PR DESCRIPTION
**Pull request by gilescope**
Fixes the question:
"Is there an easy way to just build the wasm and not build anything native? No"

No idea if this will go down like a lead balloon or whether people will like this, but here is the PR for your perusal.

If you cd into the `primitives` or the `frame` dir, you can now do `cargo check` to check all the primitives/pallet crates.
Checking wasm is now as simple as `cargo check --target waasm32-unknown-unknown`.

It also means that we don't have to list all the crates in the root Cargo.toml which I think is helpful.

It also makes it clearer which primitives/pallets support wasm and which do not.

There's lots we can't easily do with a virtual workspace so having a concrete 'workspace' crate can be handy. I don't think adding the `sp-all` and `pallet-all` crates will alter compile times but the CI can prove that out.

Curious what people think about the idea.

### CODE REVIEW

**Comment by gilescope** [comment]
(We could set the never publish Cargo.toml flag on sp-all and pallet-all)

**Comment by gilescope** [comment]
I can't see that this helps https://github.com/paritytech/substrate/issues/8974 but it certainly doesn't hinder it.

File: frame/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (expenses) [inline frame/src/lib.rs:1]:
```suggestion
/// Nothing is expected to depend upon this crate. This is purely to allow everything to be compiled to wasm. E.g. cargo check -p sp-all --no-default-features --target=wasm32-unknown-unknown
```

**Comment by thiolliere** [comment]
it feels a bit hacky to me, I feel it would be better to just write somewhere the list of crate which needs to be able to compile to wasm

**Comment by bkchr** [comment]
I see this as @thiolliere. This could also be done by some script that gets all `Cargo.toml` files that define a `std` feature.

**Comment by thiolliere** [comment]
otherwise we can still have the pallet-all and sp-all but putting them in frame/all/Cargo.toml and primitives/all/Cargo.toml without touching the Cargo.toml at the root.

**Comment by gilescope** [comment]
If you look at just the first commit @thiolliere then you will see just that. I'm happy to have them in pallet/all and primitives/all dirs.
(I can see there are people that would still want cargo fmt in the pallet dir to fmt the whole workspace.)

**Comment by thiolliere** [comment]
I mean for me the hack is to remove them from the root workspace. I think crate in substrate repo should be in the workspace. For testing in CI and other usage I think removing from workspace can have some unwanted consequences.
So I'm ok to introduce those 2 new crates, but we should keep the root workspace as it is IMHO

**Comment by bkchr** [comment]
The general question is still, why do we need this at all? And who will maintain these extra crates that serve no further purpose?

**Comment by gilescope** [comment]
Will close this for now as there's not consensus.

**Pull request closed without merging**